app/src/main/java/com/example/myphonebook/MyRecyclerViewAdapter.java

```diff
@@ -0,0 +1,78 @@
+package com.example.myphonebook;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+public class MyRecyclerViewAdapter extends RecyclerView.Adapter<MyRecyclerViewAdapter.ViewHolder> implements ContactListReceiver{
+    private ContactListProvider mData;
+    private LayoutInflater mInflater;
+    private ItemClickListener mClickListener;
+
+    MyRecyclerViewAdapter(Context context) {
+        this.mInflater = LayoutInflater.from(context);
+        this.mData = null;
+    }
+    public void setData(ContactListProvider data){
```

Comment:
`getContacts()` seems to be like an expensive operation. I suggest to make the adapter receive an array, then you won't need to use `getContact()` over and over, just once.